python/hsfs/core/arrow_flight_client.py

```diff
@@ -53,18 +53,31 @@ def __init__(self):
             try:
                 self._initialize_connection()
             except Exception as e:
-                self._is_enabled = False
-                warnings.warn(
-                    f"Could not establish connection to FlyingDuck. ({e}) "
-                    f"Will fall back to spark for this session. "
-                    f"If the error persists, you can disable FlyingDuck "
-                    f"by changing the cluster configuration (set 'enable_flyingduck'='false')."
-                )
+                self._disable(str(e))
+
+    def _disable(self, message):
+        self._is_enabled = False
+        warnings.warn(
+            f"Could not establish connection to FlyingDuck. ({message}) "
+            f"Will fall back to hive/spark for this session. "
+            f"If the error persists, you can disable FlyingDuck "
+            f"by changing the cluster configuration (set 'enable_flyingduck'='false')."
+        )
 
     def _initialize_connection(self):
         self._client = client.get_instance()
 
-        host_url = "grpc+tls://flyingduck.service.consul:5005"
+        if isinstance(self._client, client.external.Client):
+            external_domain = self._variable_api.get_loadbalancer_external_domain()
+            if external_domain == "":
+                raise Exception(
+                    "External client could not locate loadbalancer_external_domain "
+                    "in cluster configuration or variable is empty."
+                )
+            host_url = f"grpc+tls://{external_domain}:5005"
+        else:
+            host_url = "grpc+tls://flyingduck.service.consul:5005"
```

Comment:
`service.consul` can't be hardcoded. Customers have customized the end part of the domain. You should look up the variable `service_discovery_domain` on Hopsworks to retrieve the last part of the domain.

Reply:
Thanks for catching that! Just to be clear only the `consul` part of the domain is variable right?

Reply:
Yes that's correct. `service` can be hardcoded.